src/app/SecurityPanel/Credentials/CreateCredentialModal.tsx

```diff
@@ -0,0 +1,227 @@
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { CredentialAuthForm } from '@app/AppLayout/CredentialAuthForm';
+import { MatchExpressionHint } from '@app/Shared/MatchExpression/MatchExpressionHint';
+import { MatchExpressionVisualizer } from '@app/Shared/MatchExpression/MatchExpressionVisualizer';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { Target } from '@app/Shared/Services/Target.service';
+import { SearchExprService, SearchExprServiceContext } from '@app/Topology/Shared/utils';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import { evaluateTargetWithExpr, portalRoot } from '@app/utils/utils';
+import {
+  Button,
+  Card,
+  CardBody,
+  FormGroup,
+  Grid,
+  GridItem,
+  Modal,
+  ModalVariant,
+  Popover,
+  TextArea,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+
+export interface CreateCredentialModalProps {
+  visible: boolean;
+  onDismiss: () => void;
+  onPropsSave: () => void;
+}
+
+export const CreateCredentialModal: React.FunctionComponent<CreateCredentialModalProps> = ({
+  visible,
+  onDismiss,
+  onPropsSave,
+  ...props
+}) => {
+  const matchExpreRef = React.useRef(new SearchExprService());
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const alertOptions = React.useMemo(() => ({ hideActions: true }), []);
+
+  return (
+    <SearchExprServiceContext.Provider value={matchExpreRef.current}>
+      <Modal
+        appendTo={portalRoot}
+        isOpen={visible}
+        tabIndex={0} // enable keyboard-accessible scrolling
+        variant={ModalVariant.large}
+        showClose={!inProgress}
+        className="add-credential-modal"
+        onClose={onDismiss}
+        title="Store Credentials"
+        description="Create stored credentials for target JVMs. Cryostat will use these credentials to connect to Cryostat agents or target JVMs over JMX (if required)."
+      >
+        <Grid hasGutter style={{ height: '100%' }}>
+          <GridItem xl={4}>
+            <Card isFullHeight isFlat>
+              <CardBody style={{ overflow: 'auto' }}>
```

Comment:
seems like there are a lot of CardBody's with this style. Maybe better to add a class like "cryo-m-overflow-auto" and then a css rule like

.pf-c-card.cryo-m-overflow-auto {
    overflow: auto
}